include/real/exact_number.hpp

```diff
@@ -693,190 +694,322 @@ namespace boost {
+                numerator = numerator.abs();
+                denominator = denominator.abs();
 
-                // distance = (right - left) / 2
-                distance = (right - left) * half;
-                (*this) = left + distance;
-                // N/D = Q -> QD -N = 0
-                residual = (*this) * divisor - numerator;
-                if (residual == zero) {
-                    this->exponent += exponent_dif;
+                int exponent_diff = (this->exponent - 1) - (denominator.exponent - 1);
+                numerator.exponent = 1;
+                denominator.exponent = 1;
+                
+
+                if(numerator > denominator){
+                    left = one;
+                    right = numerator;
+                }else{
+                    left = zero;
+                    right = one;
+                }
+
+                length = (right - left) * half;
+                (*this) = length + left;
+
+                residual = (*this) * denominator - numerator;
+                if(residual == zero){
+                    this->exponent += exponent_diff;
                     this->positive = positive;
                     return;
                 }
 
-                // calculate the result
-                // continue the loop while we are still inaccurate (up to max precision)
-                while ((residual.abs() > min_boundary_p) && 
-                       (distance.exponent >= min_boundary_p.exponent)) {
-                    // result too small, try halfway between left and (*this) 
-                    if (residual < min_boundary_n) {
+                exact_number<T> maximum_error, neg_maximum_error;          // +-epsilon error in quotient
+                maximum_error.digits = {1};
+                maximum_error.exponent = (-1)*maximum_precision;
+                neg_maximum_error.digits = {1};
+                neg_maximum_error.exponent = (-1)*maximum_precision;
+                neg_maximum_error.positive = false;
+
+                exact_number<T> max_residual_error, neg_max_residual_error;                        // residual = (q+e)*den - num ==> residual = e*den  ( q*den = num )
+                max_residual_error = maximum_error*denominator;
+                neg_max_residual_error = max_residual_error;
+                neg_max_residual_error.positive = false;
+
+                while( (residual.abs() >= max_residual_error) && (length.exponent >= maximum_error.exponent)){
+
+                    if( (residual < neg_maximum_error) ){
                         left = (*this);
                     }
-                    // distance is halved
-                    distance = half * distance;
-                    distance.normalize();
 
-                    // truncate insignificant digits of distance
-                    while (distance.size() > maximum_precision + 1) {
-                        distance.digits.pop_back();
+                    length = length*half;
```

Comment:
try to be consistent with the style, if you leave spaces between operands (as the lines below this one) try to leave always spaces before and after operands.

Reply:
I will change that.

Reply:
actually, As I said to Vikram in his PR, try to be consistent with the previous code style. The coding style already existent in the library.

Reply:
> actually, As I said to Vikram in his PR, try to be consistent with the previous code style. The coding style already existent in the library.

Ok, I'll make sure to be consistent with library's coding style